### PR TITLE
feat: switch build to tsup, change default to esm, fix related issues

### PR DESCRIPTION
**Pull request by benbender**
This PR switches the build-tool to tsup and removes the custom build-setup. It switches the default-build to esm to align `solid-testing-library` with the rest of the ecosystem.

### CODE REVIEW

**Comment by modderme123** [comment]
Jest used to require CJS builds (because CJS can't import ESM). If you have time, it would be great to check that this still works with Jest.

**Comment by benbender** [comment]
Sure thing! I grabbed https://github.com/yeunhakim93/solid-js-app (first result for `solid-testing-library` + `jest` withe green tests + jest), removed `solid-testing-library` from `node_modules` and linked my branch. Ran without problems.

I'm unsure if there is a way to automate this without pulling in massive extra infrastructre. If you have any idea on that, I would try to provide an automated test.

**Review by modderme123 — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+    "lib": ["es2020", "dom"],
+    "module": "commonjs",
+    "target": "es2020",
     "strict": true,
-    "jsx": "preserve",
+    "esModuleInterop": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "moduleResolution": "node",
+	"jsx": "preserve",
```
Comment (atk) [inline tsconfig.json:13]: What's with the extra space?

**Review by atk — CHANGES_REQUESTED** [review]

**Comment by atk** [comment]
I have based my last updates on your work and fixed a few minor errors. Thanks for your PR, it saved me a lot of time!

**Pull request closed without merging**